lib/models/blueprint.js

```diff
@@ -770,7 +864,45 @@ let Blueprint = CoreObject.extend({
 
     this._ignoreUpdateFiles();
 
-    return finishProcessingForUninstall(fileInfos.filter(isValidFile));
+    fileInfos = fileInfos.filter(isValidFile).reduce((acc, info) => {
+      // if it's possible that this blueprint could have produced either typescript OR javascript, we have to do some
+      // work to figure out which files to delete.
+      if (this.shouldTransformTypeScript) {
+        if (this.options.typescript === true) {
+          // if the user explicitly passed `--typescript`, we only want to delete TS files, so we stick with the existing
+          // info object since it will contain a .ts outputPath (since we know this blueprint is authored in TS because
+          // of our check above)
+          acc.push(info);
+          return acc;
+        }
+
+        const jsInfo = new FileInfo({
```

Comment:
You can probably do this in the polyfill (to avoid making a require on the internal `file-info.js`)
```js
let jsInfo = Object.create(info);
Object.assign(jsInfo, { /* new stuff */ });
```